docs/authentication/saml/custom-provider.mdx

```diff
@@ -0,0 +1,123 @@
+
+## Tutorial
+
+<Steps>
+
+### Set up an enterprise connection in Clerk
+
+To create a SAML connection in Clerk:
+
+1. Navigate to the [Clerk Dashboard](https://dashboard.clerk.com).
+2. In the navigation sidebar, select **User & Authentication > [Enterprise Connections](https://dashboard.clerk.com/last-active?path=user-authentication/enterprise-connections)**.
+3. Select the **Create connection** button.
+4. You will be presented with a modal to create a new connection. Fill in the required fields and for the **Identity Provider**, select **Custom**.
+5. Next, on the **Connection** page, **do not select the Enable connection toggle just yet.** You need to fully configure your IdP first before exposing your connection to your users.
+6. Leave this page open. You will need to come back to it to complete the setup.
+
+### Create a new enterprise application in your IdP
+
+Create a new application in your IdP. In the next steps, you will configure your IdP with the settings provided by your service provider (Clerk), and you will configure Clerk with the settings provided by your IdP. So, you will need to keep both the IdP and Clerk dashboards open.
+
+### Configure your service provider
+
+To configure your service provider (Clerk), your IdP will either ask for the **Assertion Consumer Service (ACS) URL** and **Entity ID** or it will ask for the **Metadata URL**. If your IdP gives you the option to choose between the two, it is recommended to choose the **Metadata URL** as it is the quickest and most reliable way to configure your service provider.
+
+Here are what these settings mean:
+
+- **Assertion Consumer Service (ACS) URL** - This is a unique identifier for your SAML connection that your IdP application needs.
+- **Entity ID** - This is your application's URL that your IdP will redirect your users back to after they have authenticated in your IdP.
+- **Metadata URL** - This is the URL to your SAML connection's metadata file. This is the recommended way to configure your service provider.
+
+To find the values for these settings:
+
+1. In the Clerk Dashboard, find the **Service Provider configuration** section.
+2. Copy the values you need for your IdP.
+3. In your IdP dashboard, paste the values in the appropriate fields.
+
+### Configure your identity provider
+
+To configure your IdP in Clerk, scroll down to the **Identity Provider configuration** section. There are two options for configuring your IdP:
+
+- [**Metadata configuration**](#metadata-configuration) - This is where you can input the URL to your IdP's metadata file. This is the recommended way to configure your IdP, but not all IdPs support this method.
```

Comment:
There are two ways to configure the IdP using the metadata configuration. Either by a metadata url or by a metadata file. Usually n IdP offers one of the two ways